File: sol/solution/7 Kyu/544aed4c4a30184e960010f4/Solution.dart
/*
* Solution
*/
Object divisors(int number) {
  List<int> result = [];
  for (int i = 2; i <= number ~/ 2; i++) {
    if (number % i == 0) {
      result.add(i);
    }
  }
  if (result.isEmpty) {
    return "$number is prime";
  }
  return result;
}





/*
* Sample Tests
*/
// See https://pub.dartlang.org/packages/test
import "package:test/test.dart";
import "package:solution/solution.dart";

void main() {
  void doTest(int n, Object expected) {
    test('divisors($n)', () {
      expect(divisors(n), equals(expected));
    });
  }
  
  group("Sample Tests", () {
    doTest(15, [3, 5]);
    doTest(12, [2, 3, 4, 6]);
    doTest(13, '13 is prime');
  });
}

File: sol/solution/7 Kyu/544aed4c4a30184e960010f4/Solution.rs
/*
Solution
*/
fn divisors(integer: u32) -> Result<Vec<u32>, String> {
    let mut result = Vec::new();
    for i in 2..integer {
        if integer % i == 0 {
            result.push(i);
        }
    }
    if result.is_empty() {
        Err(format!("{} is prime", integer))
    } else {
        Ok(result)
    }
}





/*
Sample Tests
*/
#[cfg(test)]
mod tests {
    use super::divisors;

    fn do_test(number: u32, expected: Result<Vec<u32>, String>) {
        let actual = divisors(number);
        assert_eq!(actual, expected, "\ndivisors({number:?}): Your result (left) does not match the expected output (right)");
    }
    
    mod sample_tests {
        use super::*;

        #[test]
        fn test_15() {
	        do_test(15, Ok(vec![3,5]));
        }
        #[test]
        fn test_24() {
	        do_test(24, Ok(vec![2,3,4,6,8,12]));
        }
        #[test]
        fn test_25() {
	        do_test(25, Ok(vec![5]));
        }
        #[test]
        fn test_13() {
	        do_test(13, Err("13 is prime".into()));
        }
    }
}
